code/game/objects/items/stacks/stack_recipes.dm

```diff
@@ -506,6 +522,93 @@ var/list/datum/stack_recipe/cardboard_recipes = list (
 	new/datum/stack_recipe("empty recharge pack",           /obj/structure/vendomatpack/custom,                  4  ),
 	)
 
+/* =========================================================================
+							CLOTH RECIPES
+========================================================================= */
+
+var/list/datum/stack_recipe/cloth_recipes_by_hand = list (
+	"Simple Items",
+	new/datum/stack_recipe/cloth("Cleaning Rag",	/obj/item/weapon/reagent_containers/glass/rag,	1,	time = 20),
+	new/datum/stack_recipe/cloth("Toga",			/obj/item/clothing/under/toga,					3,	time = 50),
+	new/datum/stack_recipe/cloth("Bedsheet",		/obj/item/weapon/bedsheet/linen,				2,	time = 20),
+	)
+
+//keep in mind that tool crafting time is reduced by x0.75 with needles and x0.5 with a sewing machine, then all the way down to x0.1 with upgrades
+//a rule of thumb I settled on is 40 ticks per cloth used for the recipe
+var/list/datum/stack_recipe/cloth_recipes_with_tool = list (
```

Comment:
carpet seems like an obvious thing to sew

Reply:
I'm definitely making a note of it, that's a good idea. But same as with the map changes, I'll wait until after the update gets merged as there's lots of testing and ideas I would need to explore, that would require the other features to be here as well. (namely if we can sew carpets those should be by default of the same color as the cloth roll, but you could dye them just like clothing, except it would basically update the tile's paint_overlay, but then I'd have to add a new proc to update the overlays' shape based on how the carpet turfs smooth with their neighbors)